src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/location/LocationTest.java

```diff
@@ -0,0 +1,137 @@
+    private String systemAdminProvenanceId;
+
+    private static final String LOCATION_LIST = "locationList";
+    private static final String TEST_LOCATION_NAME = "TestLocationOne";
+    private static final String TEST_LOCATION_ID = "2001";
+    private static final String TEST_JURISDICTION1 = "Magistrates";
+    private static final String TEST_JURISDICTION2 = "Family";
+    private static final String TEST_REGION = "North West";
+
+    private static final String BASE_LOCATIONS_URL = "/locations";
+    private static final String UPLOAD_LOCATIONS_URL = BASE_LOCATIONS_URL + "/upload";
+    private static final String GET_LOCATION_BY_ID_URL = BASE_LOCATIONS_URL + "/" + TEST_LOCATION_ID;
+    private static final String GET_LOCATION_BY_NAME_URL = BASE_LOCATIONS_URL + "/name/" + TEST_LOCATION_NAME
+        + "/language/ENGLISH";
+    private static final String GET_LOCATION_BY_REGION_AND_JURISDICTION = BASE_LOCATIONS_URL + "/filter";
+    private static final String DOWNLOAD_CSV_LOCATIONS_URL = BASE_LOCATIONS_URL + "/download/csv";
+
+    @Test
+    void locationControllerHappyPathTests() throws Exception {
+        Map<String, String> headerMap = new ConcurrentHashMap<>();
+        headerMap.put(AUTHORIZATION, BEARER + accessToken);
+
+        Map<String, String> deleteHeaderMap = new ConcurrentHashMap<>();
+        deleteHeaderMap.put(AUTHORIZATION, BEARER + accessToken);
+        deleteHeaderMap.put("x-provenance-user-id", systemAdminProvenanceId);
+
+        Map<String, String> headerMapUploadLocations = new ConcurrentHashMap<>();
+        headerMapUploadLocations.put(AUTHORIZATION, BEARER + accessToken);
+        headerMapUploadLocations.put("Content-Type", MediaType.MULTIPART_FORM_DATA_VALUE);
+
+        String filePath = this.getClass().getClassLoader().getResource("location/ValidLocations.csv").getPath();
+        File csvFile = new File(filePath);
+
+        final Response responseUploadLocations = doPostRequestMultiPart(
+            UPLOAD_LOCATIONS_URL,
+            headerMapUploadLocations, LOCATION_LIST, csvFile
+        );
+
+        assertThat(responseUploadLocations.getStatusCode()).isEqualTo(OK.value());
+        assertEquals(responseUploadLocations.as(Location[].class).length, 3,
+                     "Failed to create all locations");
+        assertThat(responseUploadLocations.asString().contains(TEST_LOCATION_NAME));
+
+        final Response responseGetLocationById = doGetRequest(
+            GET_LOCATION_BY_ID_URL,
+            headerMap
+        );
+        assertThat(responseGetLocationById.getStatusCode()).isEqualTo(OK.value());
+        assertThat(responseGetLocationById.asString().contains(TEST_LOCATION_NAME));
+
+        final Response responseGetAllLocations = doGetRequest(
+            BASE_LOCATIONS_URL,
+            headerMap
+        );
+        assertThat(responseGetAllLocations.getStatusCode()).isEqualTo(OK.value());
+        assertThat(responseGetAllLocations.asString().contains(TEST_LOCATION_NAME));
```

Comment:
Using asString() in the assertion does not test the response body properly. that string can be anywhere in the response. It is better to cast the response body as the type returned by the actual endpoint so you can test the location names match up with the location ID:

List<Location> locationResults = Arrays.asList(responseUploadLocations.getBody().as(Location[].class));
assertThat(locationResults.get(0).getLocationId).isEqualTo(2001);
assertThat(locationResults.get(0).getName).isEqualTo(TEST_LOCATION_NAME);
assertThat(locationResults.get(1).getLocationId).isEqualTo(2002);
assertThat(locationResults.get(1).getName).isEqualTo("TestLocationTwo");
assertThat(locationResults.get(2).getLocationId).isEqualTo(2003);
assertThat(locationResults.get(2).getName).isEqualTo("TestLocationThree");

Reply:
This returns all courts in staging. Used filter method to assertion.